core/trino-main/src/test/java/io/trino/sql/query/TestColumnMask.java

```diff
@@ -726,4 +726,119 @@ public void testColumnMaskWithHiddenColumns()
+                new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                "clerk",
+                USER,
+                new ViewExpression(USER, Optional.empty(), Optional.empty(), "cast(regexp_replace(clerk,'(password: [^ ]+)','password: ****') as varchar(15))"));
+
+        accessControl.columnMask(
+                new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                "comment",
+                USER,
+                new ViewExpression(USER, Optional.empty(), Optional.empty(), "if(regexp_extract(clerk,'(country: [^ ]+)') IN ('country: 1'), '***', comment)"));
+
+        assertThat(assertions.query(query)).matches(expected);
+
+        // Mask "orderstatus" and "comment" using "clerk"
+        // Nothing changes for "orderstatus" and "comment" since the condition on "clerk" is not satisfied
+        accessControl.reset();
+        accessControl.columnMask(
+                new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                "clerk",
+                USER,
+                new ViewExpression(USER, Optional.empty(), Optional.empty(), "cast(regexp_replace(clerk,'(password: [^ ]+)','password: ****') as varchar(15))"));
+
+        accessControl.columnMask(
+                new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                "orderstatus",
+                USER,
+                new ViewExpression(USER, Optional.empty(), Optional.empty(), "if(regexp_extract(clerk,'(country: [^ ]+)') IN ('country: 1'), '*', orderstatus)"));
+
+        accessControl.columnMask(
+                new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                "comment",
+                USER,
+                new ViewExpression(USER, Optional.empty(), Optional.empty(), "if(regexp_extract(clerk,'(country: [^ ]+)') IN ('country: 1'), '***', comment)"));
+
+        assertThat(assertions.query(query)).matches(expected);
+
+        // Mask "comment" using "clerk" ("clerk" appears before "comment" in table definition)
+        // "comment" is masked as the condition on "clerk" is satisfied
+        accessControl.reset();
+        accessControl.columnMask(
+                new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                "clerk",
+                USER,
+                new ViewExpression(USER, Optional.empty(), Optional.empty(), "cast(regexp_replace(clerk,'(Clerk#)','***#') as varchar(15))"));
+
+        accessControl.columnMask(
+                new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                "comment",
+                USER,
+                new ViewExpression(USER, Optional.empty(), Optional.empty(), "if(regexp_extract(clerk,'([1-9]+)') IN ('951'), '***', comment)"));
+
+        assertThat(assertions.query(query))
+                .matches("VALUES (CAST('***' as varchar(79)), 'O', CAST('***#000000951' as varchar(15)))");
+
+        // Mask "comment" and "orderstatus" using "clerk" ("clerk" appears between "orderstatus" and "comment" in table definition)
```

Comment:
The order of columns shouldn't determine any mask/filter related behavior.

I don't understand this comment, but it looks like explaining why the expected behavior is what it is.
If the expected behavior was identified as dependent on the column order in the table definition, this comment should be a TODO + link to a github issue.

see also https://github.com/trinodb/trino/pull/14420#issuecomment-1265121690

Reply:
The comments on the order of column definitions do not seem related. The tests don't show any dependency between the column order and the effective semantics of the masks. @weiatwork, am I correct? What's the purpose of those comments?

Reply:
cc @kokosing

Reply:
@kasiafi That's right. That test didn't demonstrate any "column mask dependent" masking scenario, only trying to avoid the previously not so well understood behavior.

Since now I'm more confident that this behavior is problematic, I made adjustment to the test case and intentionally demo that there shouldn't be dependencies after the fix: https://github.com/trinodb/trino/pull/14420/files#diff-637b4407daa07892834452da6da9e755fd1dfb72813298df590c7fe019796249R849

Reply:
@weiatwork please see my comment https://github.com/trinodb/trino/pull/14420#issuecomment-1268086237.
If the behavior of the masks does _not_ depend on the order of column definitions, then it would be worthy to remove the related part of the comments, e.g. `("clerk" appears between "orderstatus" and "comment" in table definition)`, as they might mislead someone into thinking that there is a dependency between mask semantics and column def order.

Reply:
Currently there is dependency. Now I realize there shouldn't be any dependency among masks